pkg/awsutils/imds.go

```diff
@@ -136,6 +136,24 @@ func (imds TypedIMDS) GetMACs(ctx context.Context) ([]string, error) {
 	return list, err
 }
 
+// GetMACImdsFields returns the imds fields present for a MAC
```

Comment:
Couldn't we use `GetMACs` function and check for it. 

- Introducing yet another `GetMACImdsFields` - only to check for the `fields` might be introduced a too specialized a function for our purpose here.

Reply:
It will be preferable to use `GetMACs` to continue with coverage here.

Reply:
@orsenthil Can you add more details on using GetMACs.. So efa-only will not have any IP fields associated with them. So checking local-ipv4s field. Want to understand how we can use GetMACs here

Reply:
@Pavani-Panakanti  - you are right. I thought, we were using the `network/interfaces/macs/` and some property of that output. This is different. Sorry for the distraction.